tests/test_gae.py

```diff
@@ -112,3 +129,43 @@ def test_gae_computation(model_class, gae_lambda, gamma, num_episodes):
     # Change constant value so advantage != returns
     model.policy.constant_value = 1.0
     model.learn(rollout_size, callback=CheckGAECallback())
+
+
+@pytest.mark.parametrize("model_class", [A2C, SAC])
+@pytest.mark.parametrize("handle_timeout_termination", [False, True])
+def test_infinite_horizon(model_class, handle_timeout_termination):
+    max_steps = 8
+    gamma = 0.98
+    env = gym.wrappers.TimeLimit(InfiniteHorizonEnv(n_states=4), max_steps)
+    kwargs = {}
+    if model_class == SAC:
+        policy_kwargs = dict(net_arch=[64], n_critics=1)
+        kwargs = dict(
+            replay_buffer_kwargs=dict(handle_timeout_termination=handle_timeout_termination),
+            tau=0.5,
+            learning_rate=0.005,
+        )
+    else:
+        policy_kwargs = dict(net_arch=[64])
+        kwargs = dict(learning_rate=0.002)
+        # A2C always handle timeouts
+        if not handle_timeout_termination:
+            return
+
+    model = model_class("MlpPolicy", env, gamma=gamma, seed=1, policy_kwargs=policy_kwargs, **kwargs)
+    model.learn(1500)
+    # Value of the initial state
+    obs_tensor = model.policy.obs_to_tensor(0)[0]
+    if model_class == A2C:
+        value = model.policy.predict_values(obs_tensor).item()
+    else:
+        value = model.critic(obs_tensor, model.actor(obs_tensor))[0].item()
+    # True value (geometric serie with a reward of one at each step)
+    infinite_horizon_value = 1 / (1 - gamma)
+
+    if handle_timeout_termination:
+        # true value +/- 1
+        assert abs(infinite_horizon_value - value) < 1.0
```

Comment:
What is the choice of "1.0" here? Shouldn't `infinite_horizon_value` and `value` be close to each other (e.g. `np.isclose` check?). Sounds noisy with 1.0, even when `infinite_horizon_value` is 50.

Reply:
there are close (infinite_horizon_value ~ 50 vs value ~ 10 without proper handling ) but because the test must run fast, I reduced the number of trainings steps a lot, so the actual precision is only below 1 (in fact, for A2C, it is really close, but for SAC it depends on the seed and there is entropy... oh wait, le me set the entropy to zero)

Reply:
for reference, true value vs value when infinite horizon is properly handled or not:
```
tests/test_gae.py::test_infinite_horizon[False-SAC] 49.99999999999996 6.762823581695557
PASSED
tests/test_gae.py::test_infinite_horizon[True-A2C] 49.99999999999996 49.99942398071289
PASSED
tests/test_gae.py::test_infinite_horizon[True-SAC] 49.99999999999996 50.144901275634766
```